packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx

```diff
@@ -301,6 +305,14 @@ export class FocusZone extends React.Component<IFocusZoneProps, {}> implements I
 
     return false;
   }
+  /**
+   *  update the forceAlignment state with true/false value.
+   *  @param forceAlignmentState - if true ,focus Alignment will be recalculated depending on the currenet active element position.
+   *  @param callback - optional callback function to be executed after updating the forceAlignmentState
+   */
+  public setForceAlignmentState(forceAlignmentState: boolean, callback?: () => void) {
```

Comment:
this isn't what I was thinking. This is making the API surface really confusing. There is no need for using state at all.

You could certainly have a prop called `forceHorizontalAlignment` which is described as:

> Forces horizontal alignment in the context of vertical arrowing to always use the current selection as the reference point, rather than a center based on the last horizontal motion.

Then, when setActiveElement is called above with no param, just pass in `this.props.forceHorizontalAlignment`.